cluster/service/cluster_blackbox_test.go

```diff
@@ -48,37 +49,154 @@ func (s *ClusterServiceTestSuite) TestClustersFail() {
+	})
+}
+
+func (s *ClusterServiceTestSuite) TestRemoveIdentityToClusterLinkOK() {
+	ctx, _, reqID := tokentestsupport.ContextWithTokenAndRequestID(s.T())
+	identityID := createIdentityIDFromString(s.T())
+
+	s.T().Run("200", func(t *testing.T) {
+		clusterURL := "https://cluster.ok/"
+		r, err := recorder.New("../../test/data/cluster/unlink_identity_to_cluster", recorder.WithUnLinkIdentityToClusterRequestPayloadMatcher(clusterURL, reqID, s.saToken))
+		require.NoError(t, err)
+		defer func() { require.NoError(t, stopRecorder(r)) }()
+
+		err = s.Application.ClusterService().UnlinkIdentityFromCluster(ctx, identityID, clusterURL, rest.WithRoundTripper(r.Transport))
+		assert.NoError(t, err)
+	})
+}
+
+func (s *ClusterServiceTestSuite) TestRemoveIdentityToClusterLinkFail() {
+	ctx, _, reqID := tokentestsupport.ContextWithTokenAndRequestID(s.T())
+	identityID := createIdentityIDFromString(s.T())
+
+	s.T().Run("500", func(t *testing.T) {
+		clusterURL := "https://cluster.error/"
+		r, err := recorder.New("../../test/data/cluster/unlink_identity_to_cluster", recorder.WithUnLinkIdentityToClusterRequestPayloadMatcher(clusterURL, reqID, s.saToken))
+		require.NoError(t, err)
+		defer func() { require.NoError(t, stopRecorder(r)) }()
+
+		err = s.Application.ClusterService().UnlinkIdentityFromCluster(ctx, identityID, clusterURL, rest.WithRoundTripper(r.Transport))
+		require.EqualError(t, err, "failed to unlink identity to cluster in cluster management service. Response status: 500 Internal Server Error. Response body: oopsy woopsy")
+	})
+
+	s.T().Run("400", func(t *testing.T) {
+		clusterURL := "https://cluster.bad/"
+		r, err := recorder.New("../../test/data/cluster/unlink_identity_to_cluster", recorder.WithUnLinkIdentityToClusterRequestPayloadMatcher(clusterURL, reqID, s.saToken))
+		require.NoError(t, err)
+		defer func() { require.NoError(t, stopRecorder(r)) }()
+
+		err = s.Application.ClusterService().UnlinkIdentityFromCluster(ctx, identityID, clusterURL, rest.WithRoundTripper(r.Transport))
+		require.EqualError(t, err, "failed to unlink identity to cluster in cluster management service. Response status: 400 Bad Request. Response body: invalid identity-id")
+	})
+
+	s.T().Run("401", func(t *testing.T) {
+		clusterURL := "https://cluster.unauthorized/"
+		r, err := recorder.New("../../test/data/cluster/unlink_identity_to_cluster", recorder.WithUnLinkIdentityToClusterRequestPayloadMatcher(clusterURL, reqID, s.saToken))
+		require.NoError(t, err)
+		defer func() { require.NoError(t, stopRecorder(r)) }()
+
+		err = s.Application.ClusterService().UnlinkIdentityFromCluster(ctx, identityID, clusterURL, rest.WithRoundTripper(r.Transport))
+		require.EqualError(t, err, "failed to unlink identity to cluster in cluster management service. Response status: 401 Unauthorized. Response body: unauthorized")
+	})
+
+	s.T().Run("404", func(t *testing.T) {
```

Comment:
in its current state, the `POST /api/clusters/identities` and `DELETE /api/clusters/identities` endpoints cannot return `404 Not Found` responses (https://github.com/fabric8-services/fabric8-cluster/blob/master/design/clusters.go#L125-L149), so this test may be necessary (but I guess there's nothing wrong in supporting this case at the client level anyways...)

Reply:
it returns for DELETE look at https://github.com/fabric8-services/fabric8-cluster/blob/master/design/clusters.go#L146

Reply:
oh thanks, I missed that :/ But it's not used in the controller (https://github.com/fabric8-services/fabric8-cluster/blob/master/controller/clusters.go#L175-L198) and IMO, we shouldn't return a `404 Not Found`, but a `400 Bad Request` if the parameters in the payload are invalid, IMO

Reply:
Hmm, but it might be possible parameters in the payload are valid and there is no cluster available in DB. So, in that case, we should return 404 right?

Reply:
you mean, returning a `400 Bad Request` if the identity ID is not a valid UUID or if the cluster URL is not a parseable URL, but returning a `404 Not Found` response if there's no record matching the identityID/clusterURL ? Yeah, I guess we could do that, indeed.

Reply:
yes exactly, but we can do that only for clusterURL, because we try to find cluster using url. We don't try to find identity using identityID as we don't have any identity info in cluster. And I think it would be too much complext to check existence of identity by talking to auth.

Reply:
but for unlinking, the `404` response would mean that the identity/cluster link record was not found, right? No need to call `auth` to check that there's actually an identify matching the ID?

Reply:
yes right

Reply:
btw we are already covering this. Look at https://github.com/fabric8-services/fabric8-cluster/blob/master/controller/clusters_blackbox_test.go#L354

Reply:
ok, cool then ;)